lib/hanami/action/mime.rb

```diff
@@ -435,12 +447,33 @@ def accept
         @accept ||= @_env[HTTP_ACCEPT] || DEFAULT_ACCEPT
       end
 
-      # @since 0.1.0
+      # Checks if there is an Accept header for the current request.
+      #
+      # @return [Boolean] true if there's an Accept header to look at
+      #
+      # @since 0.7.0
       # @api private
-      def accepts
-        unless accept == DEFAULT_ACCEPT
-          best_q_match(accept, MIME_TYPES)
-        end
+      def accept_header?
+        accept != DEFAULT_ACCEPT
+      end
+
+      # Look at the Accept header for the current request and see if it
+      # matches any of the common MIME Types (see Hanami::Action::Mime#MIME_TYPES)
+      # or the custom registered ones (see Hanami::Controller::Configuration#format).
+      #
+      # @return [Nil] when the Accept header does not match any known MIME
+      # Types.
+      # @return [String] the matched MIME type for the given Accept header.
+      #
+      # @since 0.7.0
+      #
+      # @see Hanami::Action::Mime#MIME_TYPES
+      # @see Hanami::Controller::Configuration#format
+      #
+      # @api private
+      def get_mime_type_from_accept_header
+        all_types = (MIME_TYPES + configuration.format_mime_types).uniq
```

Comment:
Not sure if it's worth it, but rather than running `uniq` on a new Array, would it be worth trying out `Set` (probably `SortedSet` in this case) and seeing how that performs in comparison?